### PR TITLE
fix initialization of first_index

### PR DESCRIPTION
**Pull request by mclarsen**

### CODE REVIEW

File: src/ascent/runtimes/expressions/ascent_expression_filters.cpp [hunk 1]
```diff
@@ -1382,35 +1382,43 @@ ScalarGradient::execute()
...
+  {
     (*output)["value"] = -std::numeric_limits<double>::infinity();
     (*output)["type"] = "double";
     set_output<conduit::Node>(output);
     return;
   }
 
-  int first_index = -1, current_index = entries - 1;
-  if(execution_points) {
+  int first_index = 0, current_index = entries - 1;
```
Comment (mclarsen) [inline src/ascent/runtimes/expressions/ascent_expression_filters.cpp:1408]: This is what I changed (other than some formatting).

Comment (mlawsonca) [reply src/ascent/runtimes/expressions/ascent_expression_filters.cpp:1408]: Good catch. first_index needed to be set to -1 for a subsequent expression, but not for this one.

**Review by mlawsonca — APPROVED** [review]

**Comment by cyrush** [comment]
@mclarsen should I merge in develop (with new ci setup) and then get this merged in?

**Comment by mclarsen** [comment]
We can if we get rid of the mesh quality stuff which is not merged in vtkh.

**Comment by mclarsen** [comment]
The problem is that the mesh quality only works on completely unstructured grids, and I think in the branch, I have to turn it to unstructured. We can decide what to do about that. They fixed it in vtkm develop, but that could be a pain.

**Comment by cyrush** [comment]
got it, lets discuss in the future

**Comment by mclarsen** [comment]
closing as changes were merged in #810

**Pull request closed without merging**